mlrun/utils/notifications/notification_pusher.py

```diff
@@ -412,8 +412,17 @@ def _update_notification_status(
         sent_time: typing.Optional[datetime.datetime] = None,
         reason: typing.Optional[str] = None,
     ):
-        if run_state not in runtimes_constants.RunStates.terminal_states():
-            # we want to update the notification status only if the run is in a terminal state for BC
+        # Skip update notification state if the following conditions are met:
+        # 1. the run is not in a terminal state
+        # 2. the notification is not only for running state
+        # Skip updating because currently each notification has only one row in the db, even if it has multiple when.
+        # This means that if the notification is updated to sent for running state for example, it will not send for
+        # The terminal state
+        # TODO: Change this behavior after implementing ML-8723
+        if (
+            run_state not in runtimes_constants.RunStates.terminal_states()
+            and notification.when != ["running"]
```

Comment:
```suggestion
            and notification.when != [mlrun.common.schemas.NotificationStatus.Running]
```

Reply:
Maybe I would modify the condition to check if the when contains any terminal state instead of explicitly list of the running state, or check that the length of the list is one,  so it will be more consistent in case n new states are introduced in the future